release notes/v0.4.0.md

```diff
@@ -0,0 +1,109 @@
+
+- Added `locator.waitFor` which will wait for the element selected to meet the specified state ([#406](https://github.com/grafana/xk6-browser/pull/406))
+
+
+## Bugs fixed
+
+- Fixed zombie browser processes. Before this fix, browser processes remained active even after the test or k6 test runs. This could sometimes saturate the machine and therefore could affect the tests ([#306](https://github.com/grafana/xk6-browser/pull/306))
+
+- Fixed cleanup of the browser's temporary directory, so that it's properly removed when `Browser.close()` is called, or when the iteration or test run completes. ([#323](https://github.com/grafana/xk6-browser/pull/323), [#409](https://github.com/grafana/xk6-browser/pull/409))
+
+  This will help ensure that any private data that the browser caches to the temporary directory is deleted locally as well as when this is run in the cloud.
+
+  NOTE: There is a small chance that the temporary directory will not be cleaned and a ticket has been opened to track the next step/fix ([#403](https://github.com/grafana/xk6-browser/issues/403))
+
+- Hold modifier key (e.g. `Shift`) now works when working with `keyboard.down` and `keyboard.up` ([#325](https://github.com/grafana/xk6-browser/pull/325))
+
+  Previously the modifier key push wasn't being registered at all.
+
+- Fixed `page.content()` so that it no longer panics ([#327](https://github.com/grafana/xk6-browser/pull/327))
+
+- Fixed timeout detection ([#411](https://github.com/grafana/xk6-browser/pull/411))
+
+  Before this fix, we couldn't detect the timeouts properly. For example if we write a test where we are retrieving the contents of an element that doesn't exist, it used to return an empty string, but now it will raise a timeout error correctly. The current system is still unstable because this fix re-surfaced a data race bug that happens from time to time (happens on frame navigation). Previously reported in [#254](https://github.com/grafana/xk6-browser/issues/254)
+
+- Fixed `ElementHandle.tap()`. Before, it was performing the tap in the wrong part of the element ([#401](https://github.com/grafana/xk6-browser/pull/401))
+
+- Fixed `ElementHandle.fill()`. Before, it wasn't filling in the text in the selected input element ([#383](https://github.com/grafana/xk6-browser/pull/383))
+
+
+## Improvements
+
+- Permissions can now be granted (such as microphone and camera) to the browser ([#314](https://github.com/grafana/xk6-browser/pull/314))
+
+  This grants the new browser window permissions that would then allow tests to use those devices. You can test the change using the example in [examples/grant_permission.js](https://github.com/grafana/xk6-browser/commit/57ccb1c286e3b5fd1eb5e5d8d4109b6a3252e53f#diff-7b6083a9380e6052b6c853ab008d1993376ed01a45ed4254e1ab7e5748db8d38)
+
+- We're also removing the slow-motion option for `Frame.isChecked()` since it doesn't make sense to slow down this sort of check ([#376](https://github.com/grafana/xk6-browser/pull/376))
```

Comment:
In the future we might want to have a separate Cloud section, but this is fine for now.

```suggestion
- We're also removing the slow-motion option for `Frame.isChecked()` since it doesn't make sense to slow down this sort of check ([#376](https://github.com/grafana/xk6-browser/pull/376))

- We're working on making xk6-browser available in the k6 Cloud! The feature is still in early testing stages, but we hope to make more announcements about it soon. Stay tuned!
```